### DISCUSSION TITLE
Shellhub agent

### DISCUSSION THREAD

**Discussion by L1miter** [shellhub-io/shellhub#3637] [General]
I'm trying to start the ShellHub agent. However, after typing ./agent, I get a "no such file or directory" error.

I've copied the config.json file and modified it with the location of the shellhub.key. I've also configured the server IP address and tenant ID. I left the PATH variable untouched, but I'm still getting the same error.

I compiled the agent according to the official ShellHub guide, but it still refuses to work. Can someone tell me what I'm doing wrong?

**Comment by henrybarreto** [comment]
Hi, @L1miter. Thank you for the reporting.

Could you, please, tell us how did you compile it, and what the command you have used to running it?

Beyond that, what is your operating system and its architecture?

**Comment by L1miter** [comment]
I compiled it using the official guide on the ShellHub page: https://docs.shellhub.io/developers/agent/installing. Following the instructions for installing from source code, I used the go command from Golang and successfully built the executable. I then copied the config.json and shellhub-agent.service files from the shellhub/agent/packaging directory and moved them to another computer running Debian.

**Comment by henrybarreto** [comment]
And about the OS used to compile it; What did you use?

**Reply by L1miter** [reply]
I compiled it on debian 12.

**Comment by henrybarreto** [comment]
@L1miter, hi.

I have tried to replicate the issue, but unsuccessful; the agent ran without flaws.

## What I did?

- Run Docker container with Debian 12

```
docker run -it --name debian debian:12
```

- Install `git`
- Install `gcc` and `build-essential`
- Clone the repo as indicated in the documentation and build the agent

```
cd /root
git clone -b v0.14.2 https://github.com/shellhub-io/shellhub.git shellhub
cd shellhub/agent
go build -ldflags "-X main.AgentVersion=v0.14.2"
```

- Copy the built binary to my machine

```
docker cp debian:/root/shellhub/agent/agent .
```
- Run the agent

```
TENANT_ID=00000000-0000-4000-0000-000000000000 SERVER_ADDRESS=http://127.0.0.1 PRIVATE_KEY=/tmp/test.pk SHELLHUB_SIMPLE_USER_PASSWORD=$(openssl passwd -6) ./agent
```

Did you do something different?

**Reply by L1miter** [reply]
Following this guide worked, but I don't understand why. Could it be that it didn't work before because I compiled the agent from a previously launched shellhub server? This time, I downloaded the shellhub Git repository to another VM and compiled the agent without first launching the shellhub server.  Then, I simply connected to my shellhub server using your way of launching the agent from this VM except the simple password part, and the pending device connection was accepted and it works.

**Reply by henrybarreto** [reply]
Amazing! 

I am glad to help.

I don't think the version is the problem.  Probably, you have used the `agent` binary from the development container, what couldn't be compatible with your system as it has the same name: `agent`.
